hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/storage/ContainerProtocolCalls.java

```diff
@@ -73,6 +75,50 @@ public final class ContainerProtocolCalls  {
   private ContainerProtocolCalls() {
   }
 
+  /**
+   * Calls the container protocol to list blocks in container.
+   *
+   * @param xceiverClient client to perform call
+   * @param containerID the ID of the container to list block
+   * @param replicaIndex the index of the replica in pipeline
+   * @param startLocalID the localID of the first block to get
+   * @param count max number of blocks to get
+   * @param token a token for this block (may be null)
+   * @return container protocol list block response
+   * @throws IOException if there is an I/O error while performing the call
+   */
+  public static ListBlockResponseProto listBlock(XceiverClientSpi xceiverClient,
+      long containerID, int replicaIndex, Long startLocalID, int count,
+      Token<? extends TokenIdentifier> token) throws IOException {
+
+    ListBlockRequestProto.Builder listBlockBuilder =
+        ListBlockRequestProto.newBuilder()
+            .setCount(count);
+
+    if (startLocalID != null) {
+      listBlockBuilder.setStartLocalID(startLocalID);
+    }
+
+    String datanodeID = xceiverClient.getPipeline().getNodesInOrder()
+        .get(replicaIndex).getUuidString();
+
+    ContainerCommandRequestProto.Builder builder =
+        ContainerCommandRequestProto.newBuilder()
+            .setCmdType(Type.ListBlock)
+            .setContainerID(containerID)
+            .setDatanodeUuid(datanodeID)
+            .setListBlock(listBlockBuilder.build());
+
+    if (token != null) {
+      builder.setEncodedToken(token.encodeToUrlString());
+    }
+
+    ContainerCommandRequestProto request = builder.build();
+    ContainerCommandResponseProto response =
+        xceiverClient.sendCommand(request, getValidatorList());
```

Comment:
We may need to call `xceiverClient#sendCommandOnAllNodes` here because of the `sendCommand` node picking strategy. What do you think @umamaheswararao?

https://github.com/apache/ozone/blob/96669856ad44a7571b1dd4267f28f37055b9e4b1/hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientGrpc.java#L457-L458

https://github.com/apache/ozone/blob/96669856ad44a7571b1dd4267f28f37055b9e4b1/hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientGrpc.java#L141-L142

https://github.com/apache/ozone/blob/96669856ad44a7571b1dd4267f28f37055b9e4b1/hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientGrpc.java#L154-L155

Reply:
I just realized and it's good that server side ListBlock already handled.

I think we can create singleECBlockPipeline with the current target node, so that the strategy will just work as it will have only one node. 

Example, we did that in ECBlockOutputStreamEntry#createSingleECBlockPipeline

Reply:
> I think we can create singleECBlockPipeline with the current target node, so that the strategy will just work as it will have only one node.
> 
> Example, we did that in ECBlockOutputStreamEntry#createSingleECBlockPipeline

Good idea, thanks for the advice.